src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/IsMatch.txt

```diff
@@ -0,0 +1,117 @@
+﻿#SETUP: RegEx
+
+>> IsMatch("Hi", "H")
+false
+
+>> IsMatch("Hi", "Hi")
+true
+
+>> IsMatch("Hi", "Hi", MatchOptions.Complete)
+true
+
+>> IsMatch("Bye", "H")
+false
+
+>> IsMatch("Hi", "H", "c")
+true
+
+>> IsMatch("Hi", "H", MatchOptions.Contains)
+true
+
+>> IsMatch("Hi", "H", MatchOptions.Contains & MatchOptions.Complete)
+false
+
+>> IsMatch("Hi", "H", MatchOptions.Contains & MatchOptions.BeginWith)
+true
+
+>> IsMatch("Hi", "H", "$c")
+false
+
+>> IsMatch("Hi", "H", "c$")
+false
+
+>> IsMatch("Hi", "H", "^c")
+true
+
+>> IsMatch("Hi", "H", "c^")
+true
+
+>> IsMatch("Hi", "i", "c$")
+true
+
+>> IsMatch("Hi", "I", "c$")
+false
+
+>> IsMatch("Hi", "I", "c$i")
+true
+
+>> IsMatch("Hi", "H.*")
+true
+
+>> IsMatch("Hi", "H.*i")
+true
+
+>> IsMatch("000", Match.Digit & Match.Digit & Match.Digit)
+true
+
+>> IsMatch("JohnDoe@microsoft.com", Match.Email)
+true
+
+>> IsMatch("JohnDoe@microsoftcom", Match.Email)
+false
+
+>> IsMatch("Joe 164" & Char(10) & "Sam 208" & Char(10), "(\w+)\s(\d+)", MatchOptions.Complete & MatchOptions.Multiline)
+true
+
+>> IsMatch("Joe 164" & Char(10) & "Sam 208" & Char(10), "(\w+)\s(\d+)", MatchOptions.Complete)
+false
+
+>> IsMatch(1/0, "Hi")
+Errors: Error 9-10: Expected text. We expect text at this point in the formula.
+
+>> IsMatch(Blank(), "Hi")
+false
+
+>> IsMatch("Foo", Blank())
+Errors: Error 8-13: Expected text. We expect text at this point in the formula.|Error 0-23: The function 'IsMatch' has some invalid arguments.
+
+>> IsMatch("Foo", 17)
+Errors: Error 8-13: Expected text. We expect text at this point in the formula.|Error 0-18: The function 'IsMatch' has some invalid arguments.
+
+>> IsMatch(28, "Bar")
```

Comment:
>IsMatch(28, "Bar") [](http://example.com/codeflow?start=3&length=18)

I know PA also doesn't coerce here, but curious why don't we?
It makes sense to have a coercion...

Reply:
PA (and C#) should coerce. Now may be a good time to fix that :-)

Reply:
As discussed with Greg we won't do coercion.
Ex: IsMatch("187", 1.7) which would work in US but won't in French